api/systemNodes/initialize.js

```diff
@@ -3,14 +3,66 @@
 var self = initialize;
 module.exports = self;
 
+var async = require('async');
+
+var APIAdapter = require('../../common/APIAdapter.js');
+
 function initialize(req, res) {
   var bag = {
     reqBody: req.body,
-    resBody: {}
+    resBody: {},
+    apiAdapter: new APIAdapter(req.headers.authorization.split(' ')[1])
   };
 
   bag.who = util.format('systemNodes|%s', self.name);
   logger.info(bag.who, 'Starting');
 
-  sendJSONResponse(res, bag.resBody);
+  async.series([
+      _checkInputParams.bind(null, bag),
+      _getSystemSettings.bind(null, bag)
+    ],
+    function (err) {
+      logger.info(bag.who, 'Completed');
+      if (err)
+        return respondWithError(res, err);
+
+      sendJSONResponse(res, bag.resBody);
+    }
+  );
+}
+
+function _checkInputParams(bag, next) {
+  var who = bag.who + '|' + _checkInputParams.name;
+  logger.verbose(who, 'Inside');
+
+  if (global.config.admiralIP !== 'localhost' &&
```

Comment:
the route should do what it is intended for i.e. initialize a system node. the separation of concerns here is like this
- api provides a functionality
- client(ui/script) decides whether to use it or not
this means that this api will **always** create a system node no matter what. we will update the client to call it conditionally.